src/UserRolesService.ts

```diff
@@ -0,0 +1,66 @@
+import { Base } from "./Base";
+import { throwExpected } from "./exceptions";
+import { TeamEntityKind } from "./TeamService";
+import { Roles } from "./Permissions";
+
+interface UserSpaceRole {
+  role: Roles;
+}
+
+interface UserSpaceTeamEntity {
+  role: TeamEntityKind;
+}
+
+const rolesToTeamEntityKindMap = {
+  [Roles.Owner]: "Owner",
+  [Roles.Admin]: "Admin",
+  [Roles.Member]: "Normal",
+  [Roles.None]: "Normal",
+};
+
+const teamEntityKindToRolesMap = {
+  ["Owner" as TeamEntityKind]: Roles.Owner,
+  ["Admin" as TeamEntityKind]: Roles.Admin,
+  ["Normal" as TeamEntityKind]: Roles.Member,
+};
+
+class UserRolesService extends Base {
+  async getUserSpaceRole(spaceName: string, userName: string) {
+    const { apiEndpointAddress, fetch } = this.lensPlatformClient;
+
+    const url = `${apiEndpointAddress}/spaces/${spaceName}/users/${userName}/role`;
+
+    const json = await throwExpected(
+      async () => fetch.get(url),
+    );
+
+    const result = (json as unknown) as UserSpaceTeamEntity;
+
+    // TeamEntityKind on client are Roles. Map TeamEntityKind ro Roles.
+    return {
+      role: teamEntityKindToRolesMap[result.role],
+    };
+  }
+
+  async changeUserSpaceRole(spaceName: string, userName: string, targetRole: Roles): Promise<UserSpaceRole> {
```

Comment:
Should we disallow setting role `None`? We could at least do that on the typing level

Reply:
Setting to None is handled by the mapping function as setting to Normal/Member.

Reply:
Yes, but that seems misleading. Shouldn't we only allow setting as Roles.Member? Setting role "None" implies to me that the user should be removed from the Space.

Reply:
Added that to type check. We have a check for proper roles values on BE. We can also restrict this on client-side when selecting roles